stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,367 @@
+        comodel_name="stock.picking.type",
+        ondelete="restrict",
+        string="From operation type",
+        required=True,
+        domain="[('code', '=', 'outgoing')]",
+    )
+    from_location_src_id = fields.Many2one(
+        comodel_name="stock.location",
+        compute="_compute_from_location_src_id",
+    )
+    from_location_dest_id = fields.Many2one(
+        comodel_name="stock.location",
+        compute="_compute_from_location_dest_id",
+    )
+    to_picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        ondelete="restrict",
+        string="To operation type",
+        readonly=True,
+        domain="[('default_location_dest_id', '=', from_location_dest_id)]",
+    )
+    carrier_ids = fields.Many2many(
+        comodel_name="delivery.carrier",
+        relation="delivery_carrier_stock_warehouse_flow_rel",
+        column1="stock_warehouse_flow_id",
+        column2="delivery_carrier_id",
+        string="With carriers",
+        copy=False,
+    )
+    move_domain = fields.Char(
+        string="Source Routing Domain",
+        default=[],
+        copy=False,
+        help="Domain based on Stock Moves, to define if the "
+        "routing flow is applicable or not.",
+    )
+    delivery_steps = fields.Selection(
+        [
+            ("ship_only", "Deliver goods directly (1 step)"),
+            ("pick_ship", "Send goods in output and then deliver (2 steps)"),
+            (
+                "pick_pack_ship",
+                "Pack goods, send goods in output and then deliver (3 steps)",
+            ),
+        ],
+        "Outgoing Shipments",
+        default="ship_only",
+        required=True,
+        help="Outgoing route to follow",
+    )
+    sequence_prefix = fields.Char(
+        help=(
+            "Used to generate the default prefix of new operation types. "
+            "(e.g. 'DG' => 'WH/OUT/DG/')"
+        )
+    )
+    delivery_route_id = fields.Many2one(
+        "stock.location.route",
+        string="Delivery Route",
+        ondelete="set null",
+        readonly=True,
+    )
+    rule_ids = fields.One2many(
+        related="delivery_route_id.rule_ids",
+        string="Rules",
+    )
+    impacted_route_ids = fields.One2many(
+        comodel_name="stock.location.route",
+        compute="_compute_impacted_route_ids",
+        string="Impacted Routes",
+    )
+    warning = fields.Html(compute="_compute_warning")
+    pack_stock_loc_id = fields.Many2one(
+        "stock.location", "Packing Location", check_company=True
+    )
+    output_stock_loc_id = fields.Many2one(
+        "stock.location", "Output Location", check_company=True
+    )
+    pick_type_id = fields.Many2one(
+        "stock.picking.type", "Pick Type", check_company=True
+    )
+    pack_type_id = fields.Many2one(
+        "stock.picking.type", "Pack Type", check_company=True
+    )
+    out_type_id = fields.Many2one("stock.picking.type", "Out Type", check_company=True)
```

Comment:
Looks redundant with `to_picking_type_id`
I would drop or replace with a related field to prevent inconsistency

Reply:
I like the idea of making it a related+readonly field on `to_picking_type_id` so we get a good picture of the flow configuration in the `Technical Information` tab, like we have on a warehouse form.